lib/webmock/http_lib_adapters/net_http.rb

```diff
@@ -273,8 +273,20 @@ def self.request_signature_from_request(net_http, request, body = nil)
 
       headers = Hash[*request.to_hash.map {|k,v| [k, v]}.inject([]) {|r,x| r + x}]
 
-      headers.reject! {|k,v| k =~ /[Aa]uthorization/ && v.first =~ /^Basic / } #we added it to url userinfo
+      # If you try and make a request with headers that are symbols,
+      # Net Http raises the following NoMethodError.
+      #
+      # WebMock normalizes headers when creating a RequestSignature,
+      # and will update all headers from symbols to strings.
+      # This means that you could have a test passing with WebMock, but failing in reality.
+      header_as_symbol = headers.keys.find {|header| header.is_a? Symbol}
+      if header_as_symbol
+        raise NoMethodError.new(
```

Comment:
In response to [this comment][1]

> The only question is whether we should raise an error saying Net::HTTP doesn't except headers as symbols, which would be more useful than this error.

I think I would indeed prefer an exception making explicit that `Net::HTTP` doesn't accept symbol headers. Otherwise, if I'm a user of WebMock, I might get this exception and suspect WebMock of having a bug. It also doesn't help me much in actually fixing my code. The tradeoff, of course, is that WebMock wouldn't then be emulating `Net::HTTP` exactly. But I think situations where someone would be relying on this particular behavior being the same are incredibly rare.

[1]: https://github.com/bblimke/webmock/issues/486#issuecomment-138045396